de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/ModelAccessResource.java

```diff
@@ -10,21 +10,222 @@
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyFloat;
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyInt;
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyReference;
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyResource;
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyString;
+import de.dlr.sc.virsat.model.dvlm.Repository;
+import de.dlr.sc.virsat.project.editingDomain.VirSatTransactionalEditingDomain;
+import de.dlr.sc.virsat.server.dataaccess.CustomJsonProvider;
+import de.dlr.sc.virsat.server.dataaccess.RepositoryUtility;
+import de.dlr.sc.virsat.server.repository.RepoRegistry;
+import de.dlr.sc.virsat.server.repository.ServerRepository;
+
+/**
+ * The resource to access the VirSat data model of a server repository
+ * Provides an endpoint to access a repository
+ */
+@Path(ModelAccessResource.PATH)
 public class ModelAccessResource {
+
+	CustomJsonProvider provider;
+	
+	public static final String PATH = "/repository";
+
+	public static final String ROOT_SEIS = "seis";
+	public static final String SEI = "sei";
+	public static final String DISCIPLINES = "disciplines";
+	public static final String CONCEPTS = "concepts";
+	public static final String CA = "ca";
+	public static final String CA_AND_PROPERTIES = "caAndProperties";
+	public static final String PROPERTY = "property";
+	
+	public static final String BOOLEAN = "boolean";
+	public static final String STRING = "string";
+	public static final String INT = "int";
+	public static final String FLOAT = "float";
+	public static final String ENUM = "enum";
+	public static final String RESOURCE = "resource";
+	public static final String REFERENCE = "reference";
+	public static final String COMPOSED = "composed";
+
+	@Inject
+	public ModelAccessResource(CustomJsonProvider provider) { 
+		this.provider = provider;
+	}
+	
+	/**
+	 * Get the ServerRepository corresponding to the repoName and create a new RepoModelAccessResource
+	 * @param repoName of the repository to be accessed by the request
+	 * @return RepoModelAccessResource or null if the repo is not found
+	 */
+	@Path("{repoName}")
+	public RepoModelAccessResource getConcreteResource(@PathParam("repoName") String repoName) {
+		ServerRepository repo = RepoRegistry.getInstance().getRepository(repoName);
+		if (repo != null) {
+			VirSatTransactionalEditingDomain ed = repo.getEd();
+			provider.setEd(ed);
+			return new RepoModelAccessResource(ed.getResourceSet().getRepository());
+		}
+
+		return null;
+	}
+
+	/**
+	 * The resource to access the VirSat data model of a specific server repository
+	 * Provides the following endpoints:
+	 *   - Get roots seis
+	 *   - Get and update sei by uuid
+	 *   - Get disciplines
+	 *   - Get concepts
+	 *   - Get and update ca (with property uuids) by uuid
+	 *   - Get and update ca with properties by uuid
+	 *   - Get and update properties by uuid
+	 */
+	public static class RepoModelAccessResource {

```

Comment:
Why are these inner class and not normal classes?

Reply:
So I found it more intuitive because those classes are subresources only used in this class. But I also see no problem in changing it. Should we change it?

Reply:
I would rather split it into different files, but I'm not going to insist, we can also leave it like this.

Reply:
Ok then I will change it in https://github.com/virtualsatellite/VirtualSatellite4-Core/pull/748 after we merged this from dev so that we don't have a big merge conflict there if we change it here now